modules/support_ticket/support_ticket.module

```diff
@@ -266,6 +266,77 @@ function support_ticket_load($nid = NULL, $reset = FALSE) {
 }
 
 /**
+ * Determines what fields have changed when the support ticket is updated.
+ *
+ * @param object $support_ticket
+ *   The support ticket entity object.
+ */
+function support_ticket_entity_update($support_ticket) {
+  if (!method_exists($support_ticket, 'getFields')) {
+    return;
+  }
+
+  $fields = $support_ticket->getFields();
+  $langcodes = array_keys($support_ticket->getTranslationLanguages());
+  $original = !empty($support_ticket->original) ? $support_ticket->original: NULL;
+
+  // Don't log changes to the following fields when the support_ticket is updated.
+  // @todo: What's the cleanest way to expose this and make it configurable?
+  $ignore = array(
+    'field' => array('stid', 'uuid', 'vid', 'langcode', 'changed', 'revision_timestamp', 'revision_uid', 'revision_log', 'revision_translation_affected'),
+    'type' => array('comment'),
+  );
+  $items = array();
+  foreach ($fields as $name => $field) {
+    $type = $field->getFieldDefinition()->getType();
+    if (!in_array($name, $ignore['field']) && !in_array($type, $ignore['type'])) {
+      foreach ($langcodes as $langcode) {
+        $compare = $support_ticket->getTranslation($langcode)->get($name)->filterEmptyItems();
+        $compare_original = $original->getTranslation($langcode)->get($name)->filterEmptyItems();
+        if (!$compare->equals($compare_original)) {
+          $markup = array();
```

Comment:
To allow people to change it, I'd probably go for a view mode/display mode then rather than passing in custom settings - can add one like 'support ticket diff'?